{{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/analytics/{%if cookiecutter.use_analytics == 'y'%}serializers.py{%endif%}

```diff
@@ -0,0 +1,107 @@
+from rest_framework import serializers
+
+from {{cookiecutter.repo_name}}.account.models import User
+
+
+def get_id_context_helper(request):
+    return {
+        "ip": request.META["ANALYTICS_IP"] if "ANALYTICS_IP" in request.META else None,
+        "user_agent": request.META["HTTP_USER_AGENT"]
+        if "HTTP_USER_AGENT" in request.META
+        else None,
+    }
+
+
+def get_uuid_helper(request):
+    uuid = request.session.session_key
+
+    if not uuid:
+        request.session.create()
+        uuid = request.session.session_key
+
+    return uuid
+
+
+def get_secondary_data_helper(request, is_identifying=False):
+    if is_identifying:
+        return [
+            get_id_context_helper(request),
+            None,  # timestamp
+            None,  # integrations
+        ]
+
+    return [
+        get_id_context_helper(request),
+        request.data["timestamp"] if "timestamp" in request.data else None,
+        get_uuid_helper(request),
+        request.data["integrations"] if "integrations" in request.data else None,
+    ]
+
+
+class UserBaseSerializer(serializers.Serializer):
+    email = serializers.EmailField()
```

Comment:
I do not think this should be coming in as a piece of data - since we have the session key coming in for the request I think a much better pattern would be to see if the user is authenticated, and if so fetch email through request.user.email